### ISSUE TITLE
Focus crashes when downloading a media file on Google Pixel 6 (Android 12) 

### ISSUE THREAD

**Issue by delia-pop** [mozilla-mobile/focus-android#6346]
## Steps to reproduce
1. Open a website with media files.
2. Long tap on a link/imagine/file.
3. Tap on "Download link"/"Save image".
4. Observe the crash.
### Expected behavior
The application remains stable at all times.
### Actual behavior
The application crashes every time when trying to download a media file.
### Device information

* Android device: Google Pixel 6 (Android 12).
* Focus version: Firefox Focus Nightly 98.0 (Build 360340507 🦎 98.0a1-20220202093701).
Note that the crash was not encountered on other devices.

https://user-images.githubusercontent.com/89388888/152380122-45c93bac-2912-4928-b402-e17afc0c424c.mp4

**Comment by Mugurell** [comment]
Exception:
```
    java.lang.IllegalArgumentException: org.mozilla.focus.debug: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
    Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
        at android.app.PendingIntent.checkFlags(PendingIntent.java:375)
        at android.app.PendingIntent.getBroadcastAsUser(PendingIntent.java:645)
        at android.app.PendingIntent.getBroadcast(PendingIntent.java:632)
        at mozilla.components.feature.downloads.DownloadNotification.createPendingIntent(DownloadNotification.kt:292)
        at mozilla.components.feature.downloads.DownloadNotification.createDownloadCompletedNotification(DownloadNotification.kt:156)
        at mozilla.components.feature.downloads.AbstractFetchDownloadService.updateDownloadNotification$feature_downloads_release(AbstractFetchDownloadService.kt:374)
        at mozilla.components.feature.downloads.AbstractFetchDownloadService.updateDownloadNotification$feature_downloads_release$default(AbstractFetchDownloadService.kt:351)
        at mozilla.components.feature.downloads.AbstractFetchDownloadService.updateDownloadNotification(AbstractFetchDownloadService.kt:328)
        at mozilla.components.feature.downloads.AbstractFetchDownloadService.access$updateDownloadNotification(AbstractFetchDownloadService.kt:90)
        at mozilla.components.feature.downloads.AbstractFetchDownloadService$handleDownloadIntent$2.invokeSuspend(AbstractFetchDownloadService.kt:303)
```

@ionutbedregeanu Looks like the patch from here https://github.com/mozilla-mobile/focus-android/issues/6244 can be upstreamed to AC to then be used there and by all clients.

**Comment by delia-pop** [comment]
Verified as fixed on Firefox Focus Nightly 99.0 (Build 360450513 🦎 99.0a1-20220212115333) with Google Pixel 6 (Android 12) and Motorola Moto G30 (Android 11).
The application remains stable when downloading a media file.
